src-electron/db/query-feature.js

```diff
@@ -24,53 +24,612 @@ const dbApi = require('./db-api.js')
+  featureData,
+  endpointId,
+  missingTerms,
+  featureMap,
+  descElements
+) {
+  let featureName = featureData.name
+  let added = featureMap[featureData.code] ? true : false
+  let deviceTypeNames = featureData.deviceTypes.join(', ')
+  let result = {
+    warningMessage: '',
+    disableChange: true,
+    displayWarning: true
+  }
+  result.warningMessage = []
+
+  if (missingTerms.length > 0) {
+    let missingTermsString = missingTerms.join(', ')
+    result.warningMessage.push(
+      'On Endpoint ' +
+        endpointId +
+        ', feature ' +
+        featureName +
+        ' cannot be enabled as its conformance depends on non device type features ' +
+        missingTermsString +
+        ' with unknown values'
+    )
+  }
+
+  if (
+    (descElements.attributes && descElements.attributes.length > 0) ||
+    (descElements.commands && descElements.commands.length > 0) ||
+    (descElements.events && descElements.events.length > 0)
+  ) {
+    let attributeNames = descElements.attributes
+      .map((attr) => attr.name)
+      .join(', ')
+    let commandNames = descElements.commands
+      .map((command) => command.name)
+      .join(', ')
+    let eventNames = descElements.events.map((event) => event.name).join(', ')
+    result.warningMessage.push(
+      'On endpoint ' +
+        endpointId +
+        ', feature ' +
+        featureName +
+        ' cannot be enabled as ' +
+        (attributeNames ? 'attribute ' + attributeNames : '') +
+        (attributeNames && commandNames ? ', ' : '') +
+        (commandNames ? 'command ' + commandNames : '') +
+        ((attributeNames || commandNames) && eventNames ? ', ' : '') +
+        (eventNames ? 'event ' + eventNames : '') +
+        ' depend on the feature and their conformance are too complex to parse.'
+    )
+  }
+
+  if (
+    missingTerms.length == 0 &&
+    descElements.attributes.length == 0 &&
+    descElements.commands.length == 0 &&
+    descElements.events.length == 0
+  ) {
+    let conformance = evaluateConformanceExpression(
+      featureData.conformance,
+      featureMap
+    )
+    // change is not disabled, by default does not display warning
+    result.disableChange = false
+    result.displayWarning = false
+    // in this case only 1 warning message is needed
+    result.warningMessage = ''
+    if (conformance == 'notSupported') {
+      result.warningMessage =
+        'On endpoint ' +
+        endpointId +
+        ', feature ' +
+        featureName +
+        ' is enabled, but it is not supported for device type ' +
+        deviceTypeNames
+      result.displayWarning = added
+    }
+    if (conformance == 'provisional') {
+      result.warningMessage =
+        'On endpoint ' +
+        endpointId +
+        ', feature ' +
+        featureName +
+        ' is enabled, but it is still provisional for device type ' +
+        deviceTypeNames
+      result.displayWarning = added
+    }
+    if (conformance == 'mandatory') {
```

Comment:
These seem matter specific. I don't know where string "mandatory" is coming from, but it seems specific to Matter spec?